sys/random/random.c

```diff
@@ -54,3 +54,36 @@ void random_bytes(uint8_t *target, size_t n)
         *target++ = *random_pos++;
     }
 }
+
+uint32_t random_uint32_range(uint32_t a, uint32_t b)
+{
+    assert(a < b);
+
+    uint32_t range = b - a;
+    uint32_t highest_pow2 = (1 << bitarithm_msb(range));
+    uint32_t random = random_uint32();
+
+    /* check if range is a power of two */
+    if (!(range & (range - 1))) {
```

Comment:
I will not block this, but IMHO the code could gain readability by improving its structure. What I mean is that it essentially comes to:

1. set a proper `highest_pow2` 
2. generate a random number (fitting into range), and 
3. return it `+ a`. 

For instance something like:

```
uint32_t random_uint32_range(uint32_t a, uint32_t b)
{
    assert(a < b);

    uint32_t highest_pow2, range = b - a;
    uint8_t range_msb = bitarithm_msb(range);

    /* check if range is a power of two */
    if (!(range & (range - 1))) {
        highest_pow2 = (1 << range_msb) - 1;
    }
    else if (range_msb < 31) {
        /* leftshift for next power of two interval */
        highest_pow2 = (1 << (range_msb + 1)) -1;
    }
    else {
        /* disable modulo operation in loop below */
        highest_pow2 = UINT32_MAX;
    }
    /* get random numbers until value is smaller than range */
    do {
        random = (random_uint32() & highest_pow2);
    } while (random >= range);
    /* return random in range [a,b] */
    return (random + a);
}
```

Reply:
hope I didn't changed the logic ...

Reply:
oops forgot to define random, add change:

```
uint32_t highest_pow2, random, range = b - a;
```